raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -195,7 +216,7 @@ def handle_routechange(state, state_change):
 
 def handle_transferrefund(state, state_change):
     if state_change.sender == state.route.node_address:
```

Comment:
Please, validate *all* the fields. There is nothing checking that the amount from the refund is correct.